enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,490 @@
+
+User Workload Monitoring was designed to be primarily used by customers for monitoring their own workloads. Managed services are discouraged from using this stack since they could easily get DOS-ed by high cardinality data from end-users.
+Similarly, the configuration of the Prometheus instance itself and Alerting and Routing, would be shared with customers and other managed service owners.
+
+As a result, each MTOs currently needs to deploy its own self-developed monitoring solutions alongside its service.
+Current options like [Prometheus Operator](https://operatorhub.io/operator/prometheus) (PO) through OLM provide inital value, but they require expertise on how to build a full-fledged monitoring stack with multi-tenancy and high availability in mind (RBAC proxies, HA, Thanos Querier, Ruler etc.).
+This imposes an extra cost on teams developing and maintaining a managed service and increases time-to-market for new services.
+
+### Goals
+
+This enhancement aims to relieve the burden from managed service owners by providing a ready-made monitoring solution. Such a solution would encapsulate existing patterns already present in Platform Monitoring and User Workload monitoring.
+Furthermore, it will reduce the operational complexity of running a managed service since the Managed Tenant SREs will not need to familiarise themselves with a new monitoring stack for each managed service.
+In future it can be used to allow customers to define their monitoring stacks. UWM could use it in future underneath too. In details:
+
+* Provide a way for managed services to scrape metrics from their own services and from platform components into prometheus-compatible storage.
+* Allow managed service owners to query metrics from their services and platform components from prometheus-compatible storage.
+* Provide a way for managed services to define alerting and recording rules on their own metrics and a subset of platform metrics.
+* Provide a way for managed services to create dashboards for monitoring their services.
+* Allow aggregating portions of managed service metrics at a multi-cluster level, e.g. for reporting SLOs.
+* Allow longer retention for portions of data (e.g. calculating 1 month burn down SLO budget).
+* Allow analytics on portion of managed service data in conjuction with Telemeter data
+* End customers using UWM should not see managed service metrics.
+* Soft Goal: A foundation for next iterations mentioned in ### Long term motivation.
+
+### Long Term Motivation
+
+To avoid maintaining multiple monitoring solutions, this proposal aims to be a foundation for future monitoring enhancements. Therefore, the following goals are left out of scope for the current enhancement but are kept in mind for future iterations:
+
+* Such a solution as a community project which is easy to use & contribute to.
+* Reduce Platform Monitoring overhead and its single point of failure (SPOF) property.
```

Comment:
From a workload perspective, will this run as an INFRA workload? or a WORKER workload?

Reply:
+1 on this. Platform prom federation is not very reliable. RHOAM experienced/experiencing issues such as missing metric data at some period of time due to federation scraping timing issues or just the infra is under high load. This usually causes false-alerts in RHOAM.